agent/app/agent_unix.go

```diff
@@ -0,0 +1,57 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/cihub/seelog"
+)
+
+// control is here to help create the cgroup root '/ecs' when
+// the feature is enabled. It also aids in unit testing
+var control cgroup.Control
```

Comment:
I prefer the current pattern as it clearly separates the functionalities. It is also consistent with the way similar things are done elsewhere within the agent code-base. 

`setupTaskCgroupPrefix()` is only going to be conditionally invoked.

Reply:
Package-level variables are used to make it easier to test things without refactoring a bunch of old code.  I'd prefer not to use them when we're writing new code, if at all possible.

Not everything in this codebase is an example worth following.

Reply:
The only reason not to objectify this is because we did not want to wrap the cgroup within a platform specific object which was called out as an overkill.